sccache-dist/src/cache.rs

```diff
@@ -0,0 +1,158 @@
+use lru_disk_cache::{LruDiskCache, ReadSeek};
+use lru_disk_cache::Error as LruError;
+use lru_disk_cache::Result as LruResult;
+use ring::digest::{SHA512, Context};
+use serde_json;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::{self, File};
+use std::io::{self, BufReader, Read, Write};
+use std::path::{Path, PathBuf};
+use std::sync::Mutex;
+
+use errors::*;
+
+// TODO: possibly shouldn't be public
```

Comment:
I'm fine with TODO comments, but I'd prefer if they either get removed or if you file follow-up issues and put the issue URL in the comment.